presto-native-tests/src/test/java/com.facebook.presto.nativetests/TestDistributedEngineOnlyQueries.java

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativetests;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.type.TimeZoneKey;
+import com.facebook.presto.nativeworker.NativeQueryRunnerUtils;
+import com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestEngineOnlyQueries;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableMap;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import static com.facebook.presto.sidecar.NativeSidecarPluginQueryRunnerUtils.setupNativeSidecarPlugin;
+import static com.google.common.base.Preconditions.checkState;
+import static org.testng.Assert.assertEquals;
+
+public class TestDistributedEngineOnlyQueries
+        extends AbstractTestEngineOnlyQueries
+{
+    private static final String timeTypeUnsupportedError = ".*Failed to parse type \\[time.*";
+
+    private static final String storageFormat = "PARQUET";
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        DistributedQueryRunner queryRunner = (DistributedQueryRunner) PrestoNativeQueryRunnerUtils.createNativeQueryRunner(ImmutableMap.of(), storageFormat);
+        setupNativeSidecarPlugin(queryRunner);
+        return queryRunner;
+    }
+
+    @Override
+    protected void createTables()
+    {
+        try {
+            QueryRunner javaQueryRunner = PrestoNativeQueryRunnerUtils.createJavaQueryRunner(storageFormat);
+            NativeQueryRunnerUtils.createAllTables(javaQueryRunner, false);
+            javaQueryRunner.close();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /// TIME datatype is not supported in Prestissimo. See issue: https://github.com/prestodb/presto/issues/18844.
+    @Override
+    @Test
+    public void testTimeLiterals()
+    {
+        Session chicago = Session.builder(getSession()).setTimeZoneKey(TimeZoneKey.getTimeZoneKey("America/Chicago")).build();
+        Session kathmandu = Session.builder(getSession()).setTimeZoneKey(TimeZoneKey.getTimeZoneKey("Asia/Kathmandu")).build();
+
+        assertEquals(computeScalar("SELECT DATE '2013-03-22'"), LocalDate.of(2013, 3, 22));
+        assertQuery("SELECT DATE '2013-03-22'");
+        assertQuery(chicago, "SELECT DATE '2013-03-22'");
+        assertQuery(kathmandu, "SELECT DATE '2013-03-22'");
+
+        assertQueryFails("SELECT TIME '3:04:05'", timeTypeUnsupportedError);
```

Comment:
@pramodsatya : These are constants and should be evaluated in the co-ordinator. Any more insight into why they failed ?